src/coreclr/jit/block.cpp

```diff
@@ -292,16 +292,12 @@ bool BasicBlock::IsFirstColdBlock(Compiler* compiler) const
 //    compiler - current compiler instance
 //
 // Returns:
-//    true if the peephole optimization is enabled,
-//    and block is a BBJ_ALWAYS to the next block that we can fall through into
+//    true if block is a BBJ_ALWAYS to the next block that we can fall into
 //
 bool BasicBlock::CanRemoveJumpToNext(Compiler* compiler)
 {
     assert(KindIs(BBJ_ALWAYS));
-    const bool tryJumpOpt = compiler->opts.OptimizationEnabled() || ((bbFlags & BBF_NONE_QUIRK) != 0);
-    const bool skipJump   = tryJumpOpt && JumpsToNext() && !hasAlign() && ((bbFlags & BBF_KEEP_BBJ_ALWAYS) == 0) &&
-                          !compiler->fgInDifferentRegions(this, bbJumpDest);
-    return skipJump;
+    return (JumpsToNext() && !hasAlign() && !compiler->fgInDifferentRegions(this, bbJumpDest));
```

Comment:
nit: outermost parens are unnecessary
```suggestion
    return JumpsToNext() && !hasAlign() && !compiler->fgInDifferentRegions(this, bbJumpDest);
```